### ISSUE TITLE
Implement :cr[ewind]

### ISSUE THREAD

**Issue by adam-c-anderson** [VsVim/VsVim#2637]
**Feature category**
- Check(Type 'x') any that Applicable item:
    - [x] Vim
    - [ ] Vim plugin

If the function you request is realized by Vim or Vim plugin,
When you link the URL that is related,
We will make it easier to understand its function.

**Is your feature request related to a problem? Please describe.**
When navigating the list of errors with `:cn` and `:cp`, I want to be able to jump to the first error with the keyboard. In Vim, this would be done with the `:cr[ewind]` command.

**Describe the solution you'd like**
Implement `:cr[ewind]`

**Describe alternatives you've considered**
Mapping my own ex command, but that requires upper case which is annoying

**Comment by ricksladkey** [comment]
Good idea! I use `:cn` a lot and then use `@:` to repeat it. I sometimes have the desire to start from the top of the list so `:crewind` would be helpful.

I thought this was going to be easy but there is no `:vsc` command that I am aware of that does the resetting so I had to hunt down the tribal knowledge of how to manipulate the error programmatically.

However, researching the whole topic has kinda opened a can of worms in that we could do a lot better emulation of related functionality for the find in files dialog. If we did, we could use the same quickfix infrastructure for find next / previous, and similarly support rewinding to the first search result, which is also something I often want to do. Also, now that we can navigate to an error in the error list much more directly, we could support displaying the error in the command margin.

But all these extra ideas are quite a bit more work and I'm not yet quite read to commit to that work. But I will put them on my to do list.

**Comment by adam-c-anderson** [comment]
Awesome, thanks! Looking forward to 2.8.0 for sure now!

**Comment by ricksladkey** [comment]
Since I spent a not insignificant fraction of my life navigating errors and find results in VsVim, I got a little carried away and did all that stuff I mentioned.  See issue #2641 for screenshots.

**Comment by adam-c-anderson** [comment]
You mean you implemented `:vimgrep` and `:ln`/`:lp`/...? You beauty! 😻 

What will the commands be for first/nth match?

**Comment by ricksladkey** [comment]
The shortest commands are (as in vim): `:lne`, `:lp`, `:lfir`, `:lla`, and `:lr` for next / previous / first / last / rewind.
